src/main/c/NativeCodeSuite.cpp

```diff
@@ -0,0 +1,41 @@
+// Functions used by NativeCodeSuite.scala
```

Comment:
These should probably either be hooked up and tested from NativeCodeSuite.scala in this PR, or saved for a later PR.

Reply:
In theory, maybe.  In practice, I don't want to spend a whole lot of effort on micro-tuning the splitting of a bunch of code into individual PR's.  It's kind of inevitable that a big new chunk of
functionality lands as a big chunk of code, and we need to find a way to make it possible for
that to get steadily through the review process without adding extra difficulty.  To some extent
I think Dan has been having a similar problem with getting his off-heap/ContextRDD stuff done.

Reply:
Sure, if it's too much work, this isn't a big deal.